src/reducers/comments.spec.tsx

```diff
@@ -637,4 +641,287 @@ describe(__filename, () => {
+    });
+
+    it('fetches version comments', async () => {
+      const _getComments = jest
+        .fn()
+        .mockResolvedValue(createFakeCommentsResponse());
+      const store = configureStore();
+      const addonId = 1;
+      const versionId = 1;
+
+      const { thunk } = thunkTester({
+        createThunk: () =>
+          _fetchAndLoadComments({ _getComments, addonId, versionId }),
+        store,
+      });
+
+      await thunk();
+
+      expect(_getComments).toHaveBeenCalledWith({
+        addonId,
+        apiState: store.getState().api,
+        versionId,
+      });
+    });
+
+    it('loads comments for a version + file + line', async () => {
+      const versionId = 1;
+
+      const commentId1 = 1;
+      const commentId2 = 2;
+      const commentId3 = 2;
+
+      const fileName = 'manifest.json';
+      const line = 123;
+
+      const extComments = [commentId1, commentId2, commentId3].map((id) => {
+        return createFakeExternalComment({
+          id,
+          version: { ...fakeVersion, id: versionId },
+          filename: fileName,
+          lineno: line,
+        });
+      });
+
+      const _getComments = jest
+        .fn()
+        .mockResolvedValue(createFakeCommentsResponse(extComments));
+
+      const { dispatch, thunk } = thunkTester({
+        createThunk: () => _fetchAndLoadComments({ _getComments, versionId }),
+      });
+
+      await thunk();
+
+      expect(dispatch).toHaveBeenCalledWith(
+        actions.setComments({
+          versionId,
+          fileName,
+          line,
+          comments: extComments,
+        }),
+      );
+    });
+
+    it('loads comments for a version, file, and a line', async () => {
+      const versionId = 1;
+      const extVersion = { ...fakeVersion, id: versionId };
+
+      const commentId1 = 1;
+      const commentId2 = 2;
+      const commentId3 = 2;
+      const commentId4 = 4;
+      const commentId5 = 5;
+      const commentId6 = 6;
+
+      const fileName = 'manifest.json';
+      const line = 321;
+
+      const extComment = ({
+        version = extVersion,
+        filename = fileName,
+        lineno = line,
+        ...params
+      }: Partial<ExternalComment>) => {
+        return createFakeExternalComment({
+          version,
+          filename,
+          lineno,
+          ...params,
+        });
+      };
+
+      const versionComments = [commentId1, commentId2].map((id) => {
+        return extComment({ id, filename: null, lineno: null });
+      });
+
+      const fileComments = [commentId3, commentId4].map((id) => {
+        return extComment({ id, lineno: null });
+      });
+
+      const lineComments = [commentId5, commentId6].map((id) => {
+        return extComment({ id });
+      });
+
+      const _getComments = jest
+        .fn()
+        .mockResolvedValue(
+          createFakeCommentsResponse(
+            versionComments.concat(fileComments).concat(lineComments),
```

Comment:
Could you [`concat`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/concat#Syntax) everything with one call?

```
versionComments.concat(fileComments, lineComments)
```

Reply:
whoops, I can ✨